File: generated_tests/python/tests/audio_tests.py
import pytest
from splashkit import *
from ..helpers import *
import contextlib
class TestAudio:
    def setup_method(self):
        set_resources_path("/mnt/c/Users/Noahc/Documents/aYear_2_semester_2/TeamProject/GitHubRepo/splashkit_test_generator/resources")

    def test_audio_ready_integration():
        assert not audio_ready()
        open_audio()
        with audio_cleanup():
            assert audio_ready()
            close_audio()
            assert not audio_ready()
            
    def test_close_audio_integration():
        open_audio()
        with audio_cleanup():
            assert audio_ready()
            close_audio()
            assert not audio_ready()
            
    def test_open_audio_integration():
        open_audio()
        with audio_cleanup():
            assert audio_ready()
            close_audio()
            assert not audio_ready()
            
    def test_fade_music_in_named_integration():
        open_audio()
        with audio_cleanup():
            load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                fade_music_in_named("test_music", 1000)
                assert music_playing()
                
    def test_fade_music_in_named_with_times_integration():
        open_audio()
        with audio_cleanup():
            load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                fade_music_in_named_with_times("test_music", 2, 1000)
                assert music_playing()
                
    def test_fade_music_in_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                fade_music_in(test_music, 1000)
                assert music_playing()
                
    def test_fade_music_in_with_times_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                fade_music_in_with_times(test_music, 2, 1000)
                assert music_playing()
                
    def test_fade_music_out_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                play_music(test_music)
                fade_music_out(1000)
                delay(3000)
                assert not music_playing()
                
    def test_free_all_music_integration():
        open_audio()
        with audio_cleanup():
            load_music("test_music1", "magical_night.ogg")
            with music_cleanup():
                load_music("test_music2", "dancingFrog.wav")
                free_all_music()
                assert not has_music("test_music1")
                assert not has_music("test_music2")
                
    def test_free_music_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                play_music(test_music)
                free_music(test_music)
                assert not has_music("test_music")
                
    def test_has_music_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                assert has_music("test_music")
                free_music(test_music)
                assert not has_music("test_music")
                
    def test_load_music_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                assert test_music is not None
                assert music_valid(test_music)
                
    def test_music_filename_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                assert path_to_resource("magical_night.ogg", ResourceKind.SoundResource) == music_filename(test_music)
                
    def test_music_name_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                assert "test_music" == music_name(test_music)
                
    def test_music_named_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                named_music = music_named("test_music")
                assert test_music == named_music
                
    def test_music_playing_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                assert not music_playing()
                play_music(test_music)
                assert music_playing()
                
    def test_music_valid_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                assert music_valid(test_music)
                free_music(test_music)
                assert not music_valid(test_music)
                
    def test_music_volume_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                play_music(test_music)
                set_music_volume(0.5)
                assert 0.5 == music_volume()
                
    def test_pause_music_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                play_music(test_music)
                pause_music()
                assert music_playing()
                
    def test_play_music_named_integration():
        open_audio()
        with audio_cleanup():
            load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                play_music_named("test_music")
                assert music_playing()
                
    def test_play_music_named_with_times_integration():
        open_audio()
        with audio_cleanup():
            load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                play_music_named_with_times("test_music", 2)
                assert music_playing()
                
    def test_play_music_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                play_music(test_music)
                assert music_playing()
                
    def test_play_music_with_times_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                play_music_with_times(test_music, 2)
                assert music_playing()
                
    def test_play_music_with_times_and_volume_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                play_music_with_times_and_volume(test_music, 2, 0.75)
                assert music_playing()
                assert 0.75 == music_volume()
                
    def test_resume_music_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                play_music(test_music)
                pause_music()
                assert music_playing()
                resume_music()
                assert music_playing()
                
    def test_set_music_volume_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                play_music(test_music)
                set_music_volume(0.5)
                assert 0.5 == music_volume()
                
    def test_stop_music_integration():
        open_audio()
        with audio_cleanup():
            test_music = load_music("test_music", "magical_night.ogg")
            with music_cleanup():
                play_music(test_music)
                stop_music()
                assert not music_playing()
                
    def test_fade_all_sound_effects_out_integration():
        open_audio()
        with audio_cleanup():
            test_sound1 = load_sound_effect("test_sound1", "comedy_boing.wav")
            with sound_effect_cleanup():
                test_sound2 = load_sound_effect("test_sound2", "comedy_boing.ogg")
                play_sound_effect(test_sound1)
                play_sound_effect(test_sound2)
                fade_all_sound_effects_out(1000)
                delay(1100)
                assert not sound_effect_playing(test_sound1)
                assert not sound_effect_playing(test_sound2)
                
    def test_fade_sound_effect_out_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "breakdance.wav")
            with sound_effect_cleanup():
                play_sound_effect(test_sound)
                fade_sound_effect_out(test_sound, 1000)
                delay(3000)
                assert not sound_effect_playing(test_sound)
                
    def test_free_all_sound_effects_integration():
        open_audio()
        with audio_cleanup():
            load_sound_effect("test_sound1", "breakdance.wav")
            with sound_effect_cleanup():
                load_sound_effect("test_sound2", "comedy_boing.ogg")
                free_all_sound_effects()
                assert not has_sound_effect("test_sound1")
                assert not has_sound_effect("test_sound2")
                
    def test_free_sound_effect_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "breakdance.wav")
            with sound_effect_cleanup():
                play_sound_effect(test_sound)
                free_sound_effect(test_sound)
                assert not has_sound_effect("test_sound")
                
    def test_has_sound_effect_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "breakdance.wav")
            with sound_effect_cleanup():
                assert has_sound_effect("test_sound")
                free_sound_effect(test_sound)
                assert not has_sound_effect("test_sound")
                
    def test_load_sound_effect_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "breakdance.wav")
            with sound_effect_cleanup():
                assert test_sound is not None
                assert has_sound_effect("test_sound")
                
    def test_play_sound_effect_named_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "breakdance.wav")
            with sound_effect_cleanup():
                play_sound_effect_named("test_sound")
                assert sound_effect_playing(test_sound)
                
    def test_play_sound_effect_named_with_volume_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "breakdance.wav")
            with sound_effect_cleanup():
                play_sound_effect_named_with_volume("test_sound", 0.75)
                assert sound_effect_playing(test_sound)
                
    def test_play_sound_effect_named_with_times_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "breakdance.wav")
            with sound_effect_cleanup():
                play_sound_effect_named_with_times("test_sound", 3)
                assert sound_effect_playing(test_sound)
                
    def test_play_sound_effect_named_with_times_and_volume_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "breakdance.wav")
            with sound_effect_cleanup():
                play_sound_effect_named_with_times_and_volume("test_sound", 2, 0.75)
                assert sound_effect_playing(test_sound)
                
    def test_play_sound_effect_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "breakdance.wav")
            with sound_effect_cleanup():
                play_sound_effect(test_sound)
                assert sound_effect_playing(test_sound)
                
    def test_play_sound_effect_with_volume_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "breakdance.wav")
            with sound_effect_cleanup():
                play_sound_effect_with_volume(test_sound, 0.75)
                assert sound_effect_playing(test_sound)
                
    def test_play_sound_effect_with_times_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "SwinGameStart.wav")
            with sound_effect_cleanup():
                play_sound_effect_with_times(test_sound, 3)
                assert sound_effect_playing(test_sound)
                
    def test_play_sound_effect_with_times_and_volume_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "SwinGameStart.wav")
            with sound_effect_cleanup():
                play_sound_effect_with_times_and_volume(test_sound, 2, 0.75)
                assert sound_effect_playing(test_sound)
                
    def test_sound_effect_filename_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "SwinGameStart.wav")
            with sound_effect_cleanup():
                assert path_to_resource("SwinGameStart.wav", ResourceKind.SoundResource) == sound_effect_filename(test_sound)
                
    def test_sound_effect_name_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "SwinGameStart.wav")
            with sound_effect_cleanup():
                assert "test_sound" == sound_effect_name(test_sound)
                
    def test_sound_effect_named_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "SwinGameStart.wav")
            with sound_effect_cleanup():
                named_sound = sound_effect_named("test_sound")
                assert test_sound == named_sound
                
    def test_sound_effect_playing_named_integration():
        open_audio()
        with audio_cleanup():
            load_sound_effect("test_sound", "SwinGameStart.wav")
            with sound_effect_cleanup():
                play_sound_effect_named("test_sound")
                assert sound_effect_playing_named("test_sound")
                stop_sound_effect_named("test_sound")
                assert not sound_effect_playing_named("test_sound")
                
    def test_sound_effect_playing_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "SwinGameStart.wav")
            with sound_effect_cleanup():
                play_sound_effect(test_sound)
                assert sound_effect_playing(test_sound)
                stop_sound_effect(test_sound)
                assert not sound_effect_playing(test_sound)
                
    def test_sound_effect_valid_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "SwinGameStart.wav")
            with sound_effect_cleanup():
                assert test_sound is not None
                assert has_sound_effect("test_sound")
                
    def test_stop_sound_effect_named_integration():
        open_audio()
        with audio_cleanup():
            load_sound_effect("test_sound", "SwinGameStart.wav")
            with sound_effect_cleanup():
                play_sound_effect_named("test_sound")
                stop_sound_effect_named("test_sound")
                assert not sound_effect_playing_named("test_sound")
                
    def test_stop_sound_effect_integration():
        open_audio()
        with audio_cleanup():
            test_sound = load_sound_effect("test_sound", "SwinGameStart.wav")
            with sound_effect_cleanup():
                play_sound_effect(test_sound)
                stop_sound_effect(test_sound)
                assert not sound_effect_playing(test_sound)
                

File: generated_tests/csharp/tests/audio_tests.cs
using Xunit;
using SplashKitSDK;
using static SplashKitSDK.SplashKit;
using HttpMethod = SplashKitSDK.HttpMethod;
namespace SplashKitTests
{
    public class TestAudio
    {
        public TestAudio()
        {
            SetResourcesPath("/mnt/c/Users/Noahc/Documents/aYear_2_semester_2/TeamProject/GitHubRepo/splashkit_test_generator/resources");
        }
        [Fact]
        public void TestAudioReadyIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            Assert.True(AudioReady());
        }
        [Fact]
        public void TestCloseAudioIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            Assert.True(AudioReady());
            CloseAudio();
        }
        [Fact]
        public void TestOpenAudioIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            Assert.True(AudioReady());
            CloseAudio();
        }
        [Fact]
        public void TestFadeMusicInNamedIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            FadeMusicIn("Test Music", 1000);
            Assert.True(MusicPlaying());
        }
        [Fact]
        public void TestFadeMusicInNamedWithTimesIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            FadeMusicIn("Test Music", 2, 1000);
            Assert.True(MusicPlaying());
        }
        [Fact]
        public void TestFadeMusicInIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            FadeMusicIn(testMusic, 1000);
            Assert.True(MusicPlaying());
        }
        [Fact]
        public void TestFadeMusicInWithTimesIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            FadeMusicIn(testMusic, 2, 1000);
            Assert.True(MusicPlaying());
        }
        [Fact]
        public void TestFadeMusicOutIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            PlayMusic(testMusic);
            FadeMusicOut(1000);
            Delay(1100);
            Assert.False(MusicPlaying());
        }
        [Fact]
        public void TestFreeAllMusicIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            LoadMusic("test_music1", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            LoadMusic("test_music2", "dancingFrog.wav");
            FreeAllMusic();
            Assert.False(HasMusic("test_music1"));
            Assert.False(HasMusic("test_music2"));
        }
        [Fact]
        public void TestFreeMusicIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            PlayMusic(testMusic);
            FreeMusic(testMusic);
            Assert.False(HasMusic("Test Music"));
        }
        [Fact]
        public void TestHasMusicIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            Assert.True(HasMusic("Test Music"));
            FreeMusic(testMusic);
            Assert.False(HasMusic("Test Music"));
        }
        [Fact]
        public void TestLoadMusicIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            Assert.NotNull(testMusic);
            Assert.True(MusicValid(testMusic));
        }
        [Fact]
        public void TestMusicFilenameIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            Assert.Equal(PathToResource("magical_night.ogg", ResourceKind.SoundResource), MusicFilename(testMusic));
        }
        [Fact]
        public void TestMusicNameIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            Assert.Equal("Test Music", MusicName(testMusic));
        }
        [Fact]
        public void TestMusicNamedIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            var namedMusic = MusicNamed("Test Music");
            Assert.Equal(testMusic, namedMusic);
        }
        [Fact]
        public void TestMusicPlayingIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            Assert.False(MusicPlaying());
            PlayMusic(testMusic);
            Assert.True(MusicPlaying());
        }
        [Fact]
        public void TestMusicValidIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            Assert.True(MusicValid(testMusic));
            FreeMusic(testMusic);
            Assert.False(MusicValid(testMusic));
        }
        [Fact]
        public void TestMusicVolumeIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            PlayMusic(testMusic);
            SetMusicVolume(0.5);
            Assert.Equal(0.5, MusicVolume());
        }
        [Fact]
        public void TestPauseMusicIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            PlayMusic(testMusic);
            PauseMusic();
            Assert.True(MusicPlaying());
        }
        [Fact]
        public void TestPlayMusicNamedIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            PlayMusic("Test Music");
            Assert.True(MusicPlaying());
        }
        [Fact]
        public void TestPlayMusicNamedWithTimesIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            PlayMusic("Test Music", 2);
            Assert.True(MusicPlaying());
        }
        [Fact]
        public void TestPlayMusicIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            PlayMusic(testMusic);
            Assert.True(MusicPlaying());
        }
        [Fact]
        public void TestPlayMusicWithTimesIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            PlayMusic(testMusic, 2);
            Assert.True(MusicPlaying());
        }
        [Fact]
        public void TestPlayMusicWithTimesAndVolumeIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            PlayMusic(testMusic, 2, 0.75);
            Assert.True(MusicPlaying());
            Assert.Equal(0.75, MusicVolume());
        }
        [Fact]
        public void TestResumeMusicIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            PlayMusic(testMusic);
            PauseMusic();
            Assert.True(MusicPlaying());
            ResumeMusic();
            Assert.True(MusicPlaying());
        }
        [Fact]
        public void TestSetMusicVolumeIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            PlayMusic(testMusic);
            SetMusicVolume(0.5);
            Assert.Equal(0.5, MusicVolume());
        }
        [Fact]
        public void TestStopMusicIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testMusic = LoadMusic("Test Music", "magical_night.ogg");
            using var cleanupMusic = new MusicCleanup();
            PlayMusic(testMusic);
            StopMusic();
            Assert.False(MusicPlaying());
        }
        [Fact]
        public void TestFadeAllSoundEffectsOutIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound1 = LoadSoundEffect("Test Sound 1", "comedy_boing.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            var testSound2 = LoadSoundEffect("Test Sound 2", "comedy_boing.ogg");
            PlaySoundEffect(testSound1);
            PlaySoundEffect(testSound2);
            FadeAllSoundEffectsOut(1000);
            Delay(1100);
            Assert.False(SoundEffectPlaying(testSound1));
            Assert.False(SoundEffectPlaying(testSound2));
        }
        [Fact]
        public void TestFadeSoundEffectOutIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "breakdance.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            PlaySoundEffect(testSound);
            FadeSoundEffectOut(testSound, 1000);
            Delay(1100);
            Assert.False(SoundEffectPlaying(testSound));
        }
        [Fact]
        public void TestFreeAllSoundEffectsIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            LoadSoundEffect("Test Sound 1", "breakdance.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            LoadSoundEffect("Test Sound 2", "comedy_boing.ogg");
            FreeAllSoundEffects();
            Assert.False(HasSoundEffect("Test Sound 1"));
            Assert.False(HasSoundEffect("Test Sound 2"));
        }
        [Fact]
        public void TestFreeSoundEffectIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "breakdance.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            PlaySoundEffect(testSound);
            FreeSoundEffect(testSound);
            Assert.False(HasSoundEffect("Test Sound"));
        }
        [Fact]
        public void TestHasSoundEffectIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "breakdance.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            Assert.True(HasSoundEffect("Test Sound"));
            FreeSoundEffect(testSound);
            Assert.False(HasSoundEffect("Test Sound"));
        }
        [Fact]
        public void TestLoadSoundEffectIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "breakdance.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            Assert.NotNull(testSound);
            Assert.True(HasSoundEffect("Test Sound"));
        }
        [Fact]
        public void TestPlaySoundEffectNamedIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "breakdance.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            PlaySoundEffect("Test Sound");
            Assert.True(SoundEffectPlaying(testSound));
        }
        [Fact]
        public void TestPlaySoundEffectNamedWithVolumeIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "breakdance.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            PlaySoundEffect("Test Sound", 0.75);
            Assert.True(SoundEffectPlaying(testSound));
        }
        [Fact]
        public void TestPlaySoundEffectNamedWithTimesIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "breakdance.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            PlaySoundEffect("Test Sound", 3);
            Assert.True(SoundEffectPlaying(testSound));
        }
        [Fact]
        public void TestPlaySoundEffectNamedWithTimesAndVolumeIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "breakdance.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            PlaySoundEffect("Test Sound", 2, 0.75);
            Assert.True(SoundEffectPlaying(testSound));
        }
        [Fact]
        public void TestPlaySoundEffectIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "breakdance.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            PlaySoundEffect(testSound);
            Assert.True(SoundEffectPlaying(testSound));
        }
        [Fact]
        public void TestPlaySoundEffectWithVolumeIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "breakdance.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            PlaySoundEffect(testSound, 0.75);
            Assert.True(SoundEffectPlaying(testSound));
        }
        [Fact]
        public void TestPlaySoundEffectWithTimesIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "SwinGameStart.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            PlaySoundEffect(testSound, 3);
            Assert.True(SoundEffectPlaying(testSound));
        }
        [Fact]
        public void TestPlaySoundEffectWithTimesAndVolumeIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "SwinGameStart.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            PlaySoundEffect(testSound, 2, 0.75);
            Assert.True(SoundEffectPlaying(testSound));
        }
        [Fact]
        public void TestSoundEffectFilenameIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "SwinGameStart.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            Assert.Equal(PathToResource("SwinGameStart.wav", ResourceKind.SoundResource), SoundEffectFilename(testSound));
        }
        [Fact]
        public void TestSoundEffectNameIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "SwinGameStart.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            Assert.Equal("Test Sound", SoundEffectName(testSound));
        }
        [Fact]
        public void TestSoundEffectNamedIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "SwinGameStart.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            var namedSound = SoundEffectNamed("Test Sound");
            Assert.Equal(testSound, namedSound);
        }
        [Fact]
        public void TestSoundEffectPlayingNamedIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            LoadSoundEffect("Test Sound", "SwinGameStart.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            PlaySoundEffect("Test Sound");
            Assert.True(SoundEffectPlaying("Test Sound"));
            StopSoundEffect("Test Sound");
            Assert.False(SoundEffectPlaying("Test Sound"));
        }
        [Fact]
        public void TestSoundEffectPlayingIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "SwinGameStart.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            PlaySoundEffect(testSound);
            Assert.True(SoundEffectPlaying(testSound));
            StopSoundEffect(testSound);
            Assert.False(SoundEffectPlaying(testSound));
        }
        [Fact]
        public void TestSoundEffectValidIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "SwinGameStart.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            Assert.NotNull(testSound);
            Assert.True(HasSoundEffect("Test Sound"));
        }
        [Fact]
        public void TestStopSoundEffectNamedIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            LoadSoundEffect("Test Sound", "SwinGameStart.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            PlaySoundEffect("Test Sound");
            StopSoundEffect("Test Sound");
            Assert.False(SoundEffectPlaying("Test Sound"));
        }
        [Fact]
        public void TestStopSoundEffectIntegration() {
            OpenAudio();
            using var cleanupAudio = new AudioCleanup();
            var testSound = LoadSoundEffect("Test Sound", "SwinGameStart.wav");
            using var cleanupSoundEffect = new SoundEffectCleanup();
            PlaySoundEffect(testSound);
            StopSoundEffect(testSound);
            Assert.False(SoundEffectPlaying(testSound));
        }
    }
}
